File: 1-two-sum/1-two-sum.go
func twoSum(nums []int, target int) []int {
    numsMap := make(map[int]int)

	for i, num := range nums {
		numsMap[num] = i
	}

	for i, num := range nums {
		if j, ok := numsMap[target-num]; ok && i != j {
			return []int{i, j}

		}
	}
	return []int{}
}

File: 1-two-sum/1-two-sum.dart
class Solution {
  List<int> twoSum(List<int> nums, int target) {
    var map = <int, int>{};
    for (var i = 0; i < nums.length; i++) {
      if (map.containsKey(target - nums[i])) {
        return [map[target - nums[i]]!, i];
      } else {
        map[nums[i]] = i;
      }
    }
    return [];
  }
}
